BedrockServer.cpp

```diff
@@ -820,26 +820,52 @@ void BedrockServer::runCommand(unique_ptr<BedrockCommand>&& _command, bool isBlo
             networkLoopCount++;
             fd_map fdm;
             command->prePoll(fdm);
-            const uint64_t now = STimeNow();
-            uint64_t nextActivity = 0;
-            S_poll(fdm, max(nextActivity, now) - now);
 
-            // Timeout is five minutes unless we're shutting down or standing down, in which case it's 5 seconds.
-            // Note that BedrockCommad::postPoll sets the timeout to the command's timeout if it's lower than this value anyway,
-            // So this only has an effect if it will be shorter than the command's timeout.
-            uint64_t maxWaitMs = 5 * 60 * 1'000;
+            // Determine how long we'll wait in `poll`.
+            uint64_t maxWaitUs = 0;
+
+            // The default case is to wait until the command will time out.
+            uint64_t now = STimeNow();
+            if (now < command->timeout()) {
+                maxWaitUs = command->timeout() - now;
+            } else {
+                // The command is already timed out. This will hit the check for core.isTimedOut(command) below.
+                break;
+            }
+
+            // There are two other special cases for the time to wait.
+            if (fdm.empty()) {
+                // If there are no sockets to poll, wait 1 second.
+                // Why would there be no sockets? It's because Auth::Stripe, as a rate-limiting feature, attaches sockets to requests after their made.
```

Comment:
```suggestion
                // Why would there be no sockets? It's because Auth::Stripe, as a rate-limiting feature, attaches sockets to requests after they're made.
```